components/Seo.php

```diff
@@ -58,4 +36,152 @@ public function defineProperties()
             ]
         ];
     }
+
+    public function onRun()
+    {
+        $this->settings = Settings::instance();
+
+        if (!$this->page['viewBag']) {
+            $this->page['viewBag'] = new ViewBag();
+        }
+
+
+        if ($this->page->page->hasComponent('blogPost')) {
+            $post = $this->page['post'];
+            $properties = array_merge(
+                $this->page["viewBag"]->getProperties(),
+                $post->attributes,
+                $post->arcane_seo_options ?: []
+            );
+            $this->viewBagProperties = $properties;
+            $this->page['viewBag']->setProperties($properties);
+        } elseif (isset($this->page->apiBag['staticPage'])) {
+            $this->viewBagProperties = $this->page['viewBag'] = array_merge(
+                $this->page->controller->vars['page']->viewBag,
+                $this->page->attributes
+            );
+        } else {
+            $properties = array_merge(
+                $this->page['viewBag']->getProperties(), $this->page->settings
+            );
+
+            $this->viewBagProperties = $properties;
+            $this->page['viewBag']->setProperties($properties);
+        }
+        $this->disable_schema = $this->property('disable_schema');
+        $this->locale = App::getLocale();
+    }
+
+    public function getTitle()
+    {
+        $title = $this->viewBagProperties['title'];
+        if (isset($this->viewBagProperties['meta_title'])) {
+            $title = $this->viewBagProperties['meta_title'];
+        }
+
+        if (isset($this->viewBagProperties['meta_title[' . $this->locale . ']'])) {
+            $title = $this->viewBagProperties['meta_title[' . $this->locale . ']'];
+        }
+
+        $settings = Settings::instance();
+
+        if ($settings->site_name_position == 'prefix') {
+            $title = "{$settings->site_name} {$settings->site_name_separator} {$title}";
+        } else if ($settings->site_name_position == 'suffix') {
+            $title = "{$title} {$settings->site_name_separator} {$settings->site_name}";
+        }
+
+        return $title;
+    }
+
+    public function getDescription()
+    {
+        $description = Settings::instance()->description;
+        if (isset($this->page['description'])) {
```

Comment:
To make it more efficient we can use `elseif` or even shorter with the `??` operator.